xla/stream_executor/dnn.h

```diff
@@ -1245,6 +1245,18 @@ class VersionInfo {
   int patch_;
 };
 
+class DnnGraph {
+ public:
+  // Returns non-OK status on hard failures (incorrectly constructed graph,
+  // anything else unexpected),
+  // false on expected ones (graph is valid but not supported),
+  // true on success.
+  virtual absl::StatusOr<bool> Prepare() = 0;
```

Comment:
It seems this is triggering a compile error:

stream_executor::dnn::DnnGraph' has virtual functions but non-virtual destructor [-Werror,-Wnon-virtual-dtor]

Reply:
I don't know how to trigger it - -Werror,-Wnon-virtual-dtor did not trigger any errors with LLVM 14. Tried to fix anyway.

Reply:
Adding virtual ~DnnGraph()=default should be enough